src/ExpressiveInstaller/Resources/src/ExpressiveSymfonyContainer.php

```diff
@@ -0,0 +1,225 @@
+        }
+
+        // Inject invokables
+        foreach ($this->config['dependencies']['invokables'] as $name => $object) {
+            $container->register($name, $object);
+        }
+
+        // Inject aliases
+        foreach ($this->config['dependencies']['aliases'] as $alias => $target) {
+            $container->setAlias($alias, $target);
+        }
+
+        $container->compile();
+
+        return $container;
+    }
+
+    private function marshalDelegators(ContainerBuilder $container)
+    {
+        foreach ($this->config['dependencies']['delegators'] as $service => $delegatorNames) {
+            $factory = null;
+
+            if (isset($this->config['dependencies']['services'][$service])) {
+                // Marshal from service
+                $instance = $this->config['dependencies']['services'][$service];
+                $factory = function () use ($instance) {
+                    return $instance;
+                };
+                unset($this->config['dependencies']['services'][$service]);
+            }
+
+            if (isset($this->config['dependencies']['factories'][$service])) {
+                // Marshal from factory
+                $serviceFactory = $this->config['dependencies']['factories'][$service];
+                $factory = function () use ($service, $serviceFactory, $container) {
+                    if (is_string($serviceFactory) && class_exists($serviceFactory)) {
+                        $serviceFactory = new $serviceFactory;
+                    }
+
+                    return $serviceFactory($container, $service);
+                };
+                unset($this->config['dependencies']['factories'][$service]);
+            }
+
+            if (isset($this->config['dependencies']['invokables'][$service])) {
+                // Marshal from invokable
+                $class = $this->config['dependencies']['invokables'][$service];
+                $factory = function () use ($class) {
+                    return new $class();
+                };
+                unset($this->config['dependencies']['invokables'][$service]);
+            }
+
+            if (! is_callable($factory)) {
+                continue;
+            }
+
+            $delegatorFactory = new \App\ExpressiveSymfonyDelegatorFactory($delegatorNames, $factory);
+            $container->register($service)
+                ->addArgument(new Reference('container'))
+                ->addArgument($delegatorFactory)
+                ->addArgument($service)
+                ->setClass($service)
+                ->setFactory([\App\CallableFactory::class, 'build']);
+        }
+    }
+
+    /**
+     * @param ContainerBuilder $container
+     */
+    private function cacheContainer(ContainerBuilder $container)
+    {
+        if (null === $this->cachedContainerFile) {
+            return;
+        }
+
+        if (! (isset($this->config[static::ENABLE_CACHE]) && $this->config[static::ENABLE_CACHE])) {
```

Comment:
you should always dump the container in the cache IMO. However, you should support the debug mode of Symfony (look at how the Kernel class implements it), which automatically refreshes the cache when needed in development